Map/FloorBase.gd

```diff
@@ -0,0 +1,6 @@
+extends Resource
+class_name Floor
```

Comment:
The `Floor` class was decided against as per this comment https://github.com/Saplings-Projects/1M_sub/issues/6#issuecomment-1852966312
The current structure for the map should be a 2D array instead of an Array of floors